src/witan/phyrexian/ons_ingester.clj

```
@@ -0,0 +1,78 @@
+(ns witan.phyrexian.ons-ingester
+  (:require [clojure.java.io :as io]
+            [clojure.string :as str]
+            [clojure.set :as st]
+            [clojure.data.csv :as data-csv]
+            [clojure.core.matrix.dataset :as ds]
+            [schema.coerce :as coerce]
+            [schema.core :as s]
+            [schema.utils :as su]))
+
+(defn open-csv [filename]
+  (with-open [in-file (io/reader filename)]
+    (into []
+          (data-csv/read-csv in-file))))
+
+(defn schema-coercion [schema data]
+  (let [row ((coerce/coercer schema coerce/string-coercion-matcher) data)]
+    (if (su/error? row)
+      (vec row)
+      row)))
```

why not just return the error and then remove the error lines where you are currently doing `(filter map?)`